### PR TITLE
Revert "Deprecation errors are not suppressed anymore" #3102

### PR DESCRIPTION
**Pull request by sreichel**
### Description (*)

Please revert #3102!

With php 81. i _cant_ neither use `main` because of #3175 nor `v19` with enabled dev-mode because of deprecation errors.

### Related Pull Requests

1. See OpenMage/magento-lts#3176

### Manual testing scenarios (*)

1. use php8.1
2. enable developer mode
3. run into errors
4. ....

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All automated tests passed successfully (all builds are green)
 - [x] Add yourself to contributors list

### CODE REVIEW

**Comment by ADDISON74** [comment]
I faced this problem. the application will not run, the modification must be done manually.

However, #3102 brought advantages that we cannot deny. That's how I discovered a lot of errors/warnings in 8.2 that we didn't know about and which are mostly fixed. Unfortunately, I am the only one who approved those PRs and we've been in this situation for a long time without incorporating them into the OpenMage code faster.

**Comment by elidrissidev** [comment]
FYI there are 2 ways to prevent errors (deprecations or otherwise) from being thrown and blocking the page:

1. You just disable Developer Mode and all errors will be logged instead of thrown.
2. Update `error_reporting` directive in your `php.ini` (globally or via `.user.ini`) to only include the errors you're interested in.

Do these options not cover your case?

**Comment by fballiano** [comment]
cleaning an orphan eav attribute should be a matter of seconds.

**Comment by sreichel** [comment]
@elidrissidev disable developer mode is not an option. What was wrong with setting env `DEV_PHP_STRICT=1` to optionally show deprecation warnings? (#2882)

**Comment by fballiano** [comment]
What was wrong was already explain in the other pr.

**Comment by sreichel** [comment]
This?

> Cause it's better that people start to see this logs and fix their deprecation errors

Please let the users decide to show/hide deprecaction errors. With php8.1 there are still a lot, but i dont want to deal with them every time.

**Comment by fballiano** [comment]
Exactly, nobody wants to deal with these prs every time

**Comment by fballiano** [comment]
While we are trying to fix the few remainings of the 8.1 warnings

**Comment by sreichel** [comment]
@fballiano the env var was added for those who activly work on php81+ support - without changing behavoir for others. 

I cant work on other things while running 8.1 .... :/

**Comment by ADDISON74** [comment]
The PR description is not exactly the right one. This is not about deleting an orphaned attribute after removing an extension. That error is confirmed and we are still waiting for a solution from the one who made the recent changes in EAV.

The question that arises is why do we need another environment variable DEV_PHP_STRICT when we already have MAGE_IS_DEVELOPER_MODE? From the explanation given by @sreichel it appears that it is useful especially for the PHP8.1 version. The first variable D_P_S covers parts that the second variable M_I_D_M does not, practically they can work in parallel.

**Comment by sreichel** [comment]
> The question that arises is why do we need another environment variable DEV_PHP_STRICT when we already have MAGE_IS_DEVELOPER_MODE?

DEV-mode controlls different thing, `DEV_PHP_STRICT ` was just to show deprecation errors optionally. (escpacially for 8.1+)

Updated PR to optionally hide them.

**Comment by fballiano** [comment]
You could download this pr and apply it as a composer patch to your project

**Comment by ADDISON74** [comment]
In such situations, a democratic vote would be the best solution. As far as I am concerned, I have nothing against this PR to be included in OpenMage. There were other similar cases where developers wanted to solve their particular situations and proposed PRs in this regard and we approved them. However, Sven is an important contributor to this project and his proposals over time have found their rightful place, beyond some personal disputes. Let's not easily forget the integration of ZF1-Future, PHPStan and many others.

**Comment by sreichel** [comment]
> You could download this pr and apply it as a composer patch to your project

You can only patch files in vendor, not in project files ... no.

**Comment by ADDISON74** [comment]
In the case of ZF1-Future, we outsourced the code by linking it to the developer's repository and it ended up in the vendor directory after which we applied some patches, Composer making the work possible.

This page that was approved recently https://www.openmage.org/2023/05/01/customize-your-openmage.html, it provides a solution for those who do not see their ideas accepted in OpenMage. Now it comes the question, in order to patch the OpenMage code, do we need in vendor an openmage/magento-lts directory? I don't think this aspect has been analyzed yet.

**Comment by fballiano** [comment]
you can do it with zero problems:
https://github.com/fballiano/mcommerce/blob/main/composer.json

File: app/code/core/Mage/Core/functions.php [hunk 1]
```diff
@@ -117,6 +117,15 @@ function mageCoreErrorHandler($errno, $errstr, $errfile, $errline)
         return false;
     }
 
+    // Suppress deprecation warnings on PHP 7.x
+    // set environment variable DEV_PHP_STRICT to 0 will hide E_DEPRECATED errors
+    if ((isset($_ENV['DEV_PHP_STRICT']) && $_ENV['DEV_PHP_STRICT'] == '0')
+        && $errno == E_DEPRECATED
+        && version_compare(PHP_VERSION, '7.0.0', '>=')
```
Comment (fballiano) [inline app/code/core/Mage/Core/functions.php:124]: at this point it would be better to mage developer mode = error reporting

**Comment by fballiano** [comment]
> I hope we don't end up proposing patches to everyone who wants basic features.

was this ever done? never. so, in this case, it was done for a reason.

**Comment by sreichel** [comment]
> you can do it with zero problems:

Locally i work with `git pull` and have no `openmage/magento-lts` dependency.

**Comment by sreichel** [comment]
Can we just restore a working behavoir? (At least for `v19`)

- deprecation warnings were hidden all the time ... (b/c they do not harm)
- a ENV var was added to show/hide them
- everything got removed and everyone faces deprecation warnings for php81

I REALLY dont understand what we are discussing bout. 

Actually the best way to work arround it, is not to use php81 at all.

**Comment by Flyingmana** [comment]
is there a problem with the mentioned "error_reporting" approach`? Do we overwrite the ini setting in OpenMage? Because then we should make this configurable for Dev Mode, as people might want different levels, sometimes without error reporting, sometimes even without notices.
Personally I did always hardcode changed error reporting somewhere 🤔

**Comment by sreichel** [comment]
> as people might want different levels, sometimes without error reporting, sometimes even without notices.

Thats it, This change killed the possibilty to hide/show deprecation warnings ...

**Comment by Flyingmana** [comment]
So I tested it out, and I could confirm the error_reporting is not overwritten from the dev mode, so setting in the php.ini `error_reporting = 24575` (which is E_ALL & ~E_DEPRECATED) prevents your reported issue.
Therefor I vote for closing this PR

**Comment by sreichel** [comment]
PR was w/o adjusting php.ini ...

**Comment by sreichel** [comment]
Thank for wasting my time.

**Pull request closed without merging**